### ISSUE TITLE
Updater overwrites gitconfig file

### ISSUE THREAD

**Issue by mbcmike** [ForkIssues/TrackerWin#530]
I am using AWS CodeCommit. AWS CodeCommit requires changes to the gitconfig file as following:
```
[credential]
usehttppath=true
helper=!aws codecommit credential-helper $@
```
Everytime Fork updates, it rewrites the gitconfig file back to it's original state and I have to go back in and change it. It would be great if it didn't do that.

**Comment by DanPristupov** [comment]
**Which** git config?

---------

> Everytime Fork updates

Not every time, but only when we upgrade git (for example 1.40-> 1.41). It happened during 1.41.0 -> 1.41.1 because git 2.23 had a bug in the config and I had to rewrite to.

This is the *instance* config I think you should not edit it.

**Comment by mbcmike** [comment]
Sorry  /mingw64/etc/gitconfig

**Comment by DanPristupov** [comment]
I suppose you mean the one in the `%localappdata\fork\gitinstance\` directory?

How did you edit it?

Why didn't you use the `git config --global` location?

You can check where the values are defined like following (should be `C:/Users/username/.gitconfig`)
```
git config --show-origin credential.usehttppath
git config --show-origin credential.helper
```

```
git config --global credential.usehttppath true
git config --global credential.helper !aws codecommit credential-helper $@
```

**Comment by mbcmike** [comment]
I originally posted about this issue in #350. It seemed that fork was not using my configuration I had setup in git --global. So it was suggested to modify the  /mingw64/etc/gitconfig instead. Which worked just fine, until the latest updates I had to redo the work.

**Comment by DanPristupov** [comment]
I see now. Let me think.

**Comment by DanPristupov** [comment]
OK, it looks like `--system` config values override the `--global` ones. And since the `--system` one has `credential.helper = manager` you don't have a way other then modify it there. I think there's nothing we can change.

Returning to the original issue of this topic, the complaint is absolutely legit. When I update git instance next time, I will copy and restore the `gitconfig` as a part of the upgrade process. It would be nice to hear your feedback after that.

P.S. I think I can close #350

**Comment by mbcmike** [comment]
Yes you can close #350. I will absolutely give you back feedback. Huge fan of Fork and all the work that you do!

**Comment by DanPristupov** [comment]
I just released Fork 1.0.42.**6**. It is a hotfix which upgrades git to 2.24.1 (some serious security vulnerabilities were found in the previous versions).

Fork should handle the migration correctly and also restore gitconfig of the previous instance.

**Comment by mbcmike** [comment]
I don't think it worked. When I opened the console in fork and did a "git config --system -l", I didn't see my codecommit helper. I had to do a "git config --system -e" to setup codecommit. Now it works again.
